src/bsoncxx/array/value.cpp

```diff
@@ -23,6 +23,13 @@ namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 namespace array {
 
+namespace {
+const auto nop_deleter = [](uint8_t*) {};
+}  // namespace
+
+value::value() noexcept : _data(nullptr, nop_deleter), _length() {
```

Comment:
Does this technically involve static initialization order fiasco's if default constructed views are created during dynamic initialization?

Free function is possibly safer?